datatree/tests/test_datatree.py

```diff
@@ -440,6 +440,38 @@ def test_setitem_dataarray_replace_existing_node(self):
         xrt.assert_identical(results.to_dataset(), expected)
 
 
+class TestMoveNode:
+    dt = DataTree.from_dict(
+        d={
+            "/": xr.Dataset({"name": "Hobert"}),
+            "/Jake": xr.Dataset({"name": "Jake"}),
+            "/Bragg": xr.Dataset({"name": "Bragg"}),
+            "/Bragg/Kyle": xr.Dataset({"name": "Noah"}),
+            "/Kai": xr.Dataset({"name": "Kai"}),
+        },
+        name="TestMove",
+    )
+
+    def test_move_to_root(self):
+        new_dt = self.dt.move("/Bragg/Kyle", "/Kyle")
+        xrt.assert_equal(
+            self.dt["/Bragg/Kyle"].to_dataset(), new_dt["/Kyle"].to_dataset()
+        )
+        assert new_dt.get("/Bragg/Kyle", None) is None
+
+    def test_move_under_subnode(self):
+        new_dt = self.dt.move("/Jake", "/Bragg/Kyle/Jake")
+        xrt.assert_equal(
+            self.dt["/Jake"].to_dataset(), new_dt["/Bragg/Kyle/Jake"].to_dataset()
+        )
```

Comment:
There are assert functions for comparing two datatrees that you can use to check that the whole tree is as expected.

Generally you want tests to take the form

```python
def test_thing():
    original = ...
    expected_result = ...
    result = dt.thing()
    assert_datatree_equal(result, expected_result)
```
```

Reply:
I haven't found a method `assert_datatree_equal` to compare if two datatrees are equal